packages/plugin-commands-installation/src/recursive.ts

```diff
@@ -253,6 +259,8 @@ export default async function recursive (
     : Object.keys(opts.selectedProjectsGraph).sort()
 
   const limitInstallation = pLimit(opts.workspaceConcurrency ?? 4)
+  const patternedInput = input.filter(i => i.includes('*'))
+  const updateMatch = cmdFullName === 'update' && patternedInput.length ? matcher(patternedInput) : null
```

Comment:
These two lines can be moved above line 167. In that case, you can remove the duplicated lines at 180-181